File: sistema-academico/src/main/java/com/universidad/sistema_academico/entity/Nota.java
package com.universidad.sistema_academico.entity;

import jakarta.persistence.*;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@Entity
@Table(name = "notas")
public class Nota {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @OneToOne(optional = false)
    @JoinColumn(name = "detalle_matricula_id", unique = true)
    private DetalleMatricula detalle;

    @Column(precision = 4, scale = 2)
    private BigDecimal parcial1;

    @Column(precision = 4, scale = 2)
    private BigDecimal parcial2;

    @Column(precision = 4, scale = 2)
    private BigDecimal notaFinal;

    @Column(precision = 4, scale = 2)
    private BigDecimal promedio;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 15)
    private EstadoNota estado = EstadoNota.PENDIENTE;

    private LocalDateTime fechaActualizacion;
}

File: frontend/src/types/nota.ts
import type { DetalleMatricula } from './detalleMatricula'

export type EstadoNota = 'PENDIENTE' | 'APROBADO' | 'DESAPROBADO'

export interface Nota {
  id: number
  detalle: DetalleMatricula
  parcial1?: number | null
  parcial2?: number | null
  notaFinal?: number | null
  promedio?: number | null
  estado: EstadoNota
  fechaActualizacion?: string | null
}

export const ESTADO_NOTA_LABELS: Record<EstadoNota, string> = {
  PENDIENTE: 'Pendiente',
  APROBADO: 'Aprobado',
  DESAPROBADO: 'Desaprobado',
}
